source/engines/rollEngine.js

```diff
@@ -1,548 +1,771 @@
-							highIndex = i;
-						}
-					}
-					dieMap[highIndex].keep = false;
-					skipIndices.push(highIndex);
-					repeats--;
-				}
-				break;
-		}
-		return `(${dieMap.map(dieInfo => dieInfo.keep ? dieInfo.roll.toString(frac) : `~~${dieInfo.roll.toString(frac)} dropped~~${frac ? '0/0' : '0'}`).join(' + ')})`;
-	}
-}
-
-class ResultBundle {
-	#resultset
-	#extraText
-
-	constructor(rs, et = ""){
-		var resultSet;
-		if (rs instanceof ResultSet) { // If it is a result set, just use it as-is
-			resultSet = rs;
-		} else { // Parse the string into a roll set
-			resultSet = ResultSet.parse(rs);
-		}
-		this.#resultset = resultSet;
-		this.#extraText = et;
-	}
-
-	getResultSet() {
-		return this.#resultset;
-	}
-
-	getExtraText() {
-		return this.#extraText;
-	}
-
-	toString(frac = false, simple = false) {
-		if (simple && frac) {
-			return `${this.#resultset.getResults()}/${this.#resultset.getMaxResults()}${this.#extraText ? ` ${this.#extraText}` : ''}`;
-		} else if (simple && !frac) {
-			return `${this.#resultset.getResults()}${this.#extraText ? ` ${this.#extraText}` : ''}`;
-		} else if (!simple && frac) {
-			return `${this.#resultset.toString(frac)} = ${this.#resultset.getResults()}/${this.#resultset.getMaxResults()}${this.#extraText ? ` ${this.#extraText}` : ''}`;
-		} else {
-			return `${this.#resultset.toString(frac)} = ${this.#resultset.getResults()}${this.#extraText ? ` ${this.#extraText}` : ''}`;
-		}
-	}
-}
-
-function parseRoll(input) { //This needs some wild revision before it'll work
-	input = input.replace(/\(/g,' ( ').replace(/\)/g,' ) ')
-	var splitInput = input.split(' ');
-	var rollArray = splitInput.reduce((acc, cur) => {
-		if (acc[acc.length-1] == '║' || cur == '') {
-			return acc;
-		} else if (/[\+\-\*]/.test(cur) || /([0-9]+|[0-9]+d[0-9]+([dk][lh]?[0-9]+)?)([\+\-\*]([0-9]+|[0-9]+d[0-9]+([dk][lh]?[0-9]+)?))*$/.test(cur) || /\(/.test(cur) || /\)/.test(cur)) {
-			acc.push(cur);
-			return acc;
-		} else { //stop on the first non-roll
-			acc.push('║');
-			return acc;
-		}
-	}, []);
-	rollArray[rollArray.length-1] == '║' ? rollArray.pop() : ()=>{};
-	var textArray = splitInput.reduceRight((acc, cur) => {
-		if (acc[0] == '║' || cur == '') {
-			return acc;
-		} else if (/([0-9]+|[0-9]+d[0-9]+([dk][lh]?[0-9]+)?)([\+\-\*]([0-9]+|[0-9]+d[0-9]+([dk][lh]?[0-9]+)?))*$/.test(cur) || /[0-9]+$/.test(cur) || /\)/.test(cur)) {
-			acc.unshift('║');
-			return acc;
-		} else {
-			acc.unshift(cur);
-			return acc;
-		}
-	}, []);
-	textArray[0] == '║' ? textArray.shift() : ()=>{};
-	return new ResultBundle(rollArray.join(''),textArray.join(' '));
-}
-
-function getRollString(input, frac = false, simple = false) {
-	return parseRoll(input).toString(frac, simple);
-}
-
-module.exports = {parseRoll, getRollString}
+class Result {
+	#result;
+
+	constructor(result) {
+		this.#result = result;
+	}
+
+	getResult() {
+		return this.#result;
+	}
+
+	toString() {
+		return `${this.#result}`;

```

Comment:
On the assumption that `Result.prototype.result` is a `string | number`, I think it would be neater to `Result.toString()` in-line since both String and Number have good `.toString()` methods of their own.

Reply:
This is more of a pattern thing, so that the means by which the string is constructed in this simple case matches how it is done in the more complex cases later. (I also have another comment about crossing wires between `getResult` and `toString` that is only partially relevant here.)